src/Symfony/Component/Security/Http/MagicLink/MagicLinkTokenStorageInterface.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\MagicLink;
+
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * A class that is able to store, find and invalidate tokens used in magic links.
+ *
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ */
+interface MagicLinkTokenStorageInterface
+{
+    /**
+     * Stores all the token information using the selector as the lookup key.
+     *
+     * The key that is sent to to the user will normally be a combination
+     * of selector + the un-hashed verifier.
+     */
+    public function storeToken(string $selector, string $hashedVerifier, UserInterface $user, \DateTimeInterface $expiresAt): void;
+
+    /**
+     * Find and return the token information, wrapped in an instance of AuthenticatableTokenInterface.
```

Comment:
this phpdoc looks wrong to me. It returns a StoredMagicLinkTokenInterface, not a AuthenticatableTokenInterface